src/twisted/newsfragments/10249.feature

```diff
@@ -0,0 +1,5 @@
+The ``all_non_platform`` extra has been renamed to ``all``,
```

Comment:
all currently pulls in `    %(test)s` which is probably not what users of twisted want

Reply:
eg if you do `pip install twisted[all]` it will also install `cython-test-exception-raiser ~= 1.0` which nobody needs and I'll get angry letters about it

Reply:
Apparently this has not been documented, need we add a news fragment for it or we can pretend that the `test` extra never existed?

> The plan is to remove the `test` extra requirement and skip tests that depend on then [sic]